src/state/network/updater.tsx

```diff
@@ -9,15 +9,40 @@ import { useNetworkId } from './hooks'
 import { updateWeb3Provider } from 'api/web3'
 import { web3 } from 'apps/explorer/api'
 
-function getNetworkId(network: string | undefined): Network {
-  switch (network) {
-    case 'rinkeby':
-      return Network.Rinkeby
-    case 'xdai':
-      return Network.xDAI
-    default:
-      return Network.Mainnet
+const MAINNET_PREFIX = ''
+const NETWORK_PREFIXES_RAW: [Network, string][] = [
+  [Network.Mainnet, ''],
+  [Network.xDAI, 'xdai'],
+  [Network.Rinkeby, 'rinkeby'],
+]
+const PREFIX_BY_NETWORK_ID: Map<Network, string> = new Map(NETWORK_PREFIXES_RAW)
```

Comment:
why not just use the `enum` here instead of all these `Maps`? 

```ts
enum ChainId {
    MAINNET = 1,
    ROPSTEN = 3,
    RINKEBY = 4,
    GÖRLI = 5,
    KOVAN = 42,
    XDAI = 100
}

const network = 1
const networkName = "MAINNET"

console.log(ChainId[network])
// MAINNET
console.log(ChainId[networkName])
// 1
```